pkg/skaffold/docker/parse.go

```diff
@@ -214,15 +214,16 @@ func extractCopyCommands(nodes []*parser.Node, onlyLastImage bool, cfg Config) (
 		switch node.Value {
 		case command.From:
 			from := fromInstruction(node)
+			if from.as != "" {
+				// Stage names are case insensitive
+				stages[strings.ToLower(from.as)] = true
+			}
+
 			if from.image == "" {
 				// some build args like artifact dependencies are not available until the first build sequence has completed.
 				// skip check if there are unavailable images
 				continue
```

Comment:
immediately continuing with the next iteration was wrong here; we need to save the instruction's `AS` value also to `stages` map.